core/camel-core/src/test/java/org/apache/camel/processor/SagaTimeoutTest.java

```diff
@@ -81,6 +110,29 @@ public void configure() throws Exception {
                         .compensation("mock:compensate").completion("mock:complete")
                         .to("mock:end");
 
+                from("direct:saga-multi-participants")
+                        .process(exchange -> {
+                            exchange.getMessage().setHeader("id", UUID.randomUUID().toString());
+                        })
+                        .saga()
+                        .propagation(SagaPropagation.REQUIRES_NEW)
+                        .to("direct:service1")
+                        .to("direct:service2");
+
+                from("direct:service1")
+                        .saga().option("id", header("id"))
+                        .propagation(SagaPropagation.MANDATORY).timeout(100, TimeUnit.MILLISECONDS)
+                        .compensation("mock:compensate").completion("mock:complete")
+                        .process(exchange -> {
+                            Thread.sleep(300L, 0);
+                        })
```

Comment:
You could use the `delay` EIP instead https://camel.apache.org/components/3.20.x/eips/delay-eip.html

Reply:
Nice catch! I will change to use `deplay`. Thanks!

Reply:
Done